indra/llwindow/llkeyboard.cpp

```diff
@@ -273,12 +276,49 @@ BOOL LLKeyboard::handleTranslatedKeyUp(KEY translated_key, U32 translated_mask)
 		mKeyUp[translated_key] = TRUE;
 		handled = mCallbacks->handleTranslatedKeyUp(translated_key, translated_mask);
 	}
-	
+
 	LL_DEBUGS("UserInput") << "keyup -" << translated_key << "-" << LL_ENDL;
 
 	return handled;
 }
 
+BOOL LLKeyboard::handleKeyDown(const U16 key, const U32 mask)
+{
+	U32	translated_mask = updateModifiers(mask);
+
+	KEY		translated_key = 0;
+	BOOL	handled = FALSE;
+	if(translateKey(key, &translated_key))
+	{
+		handled = handleTranslatedKeyDown(translated_key, translated_mask);
+	}
+    if (!handled)
+    {
+        LLGameControl::onKeyDown(translated_key, translated_mask);
+    }
+
+	return handled;
+}
+
+
+BOOL LLKeyboard::handleKeyUp(const U16 key, const U32 mask)
+{
+	U32	translated_mask = updateModifiers(mask);
+
+	KEY		translated_key = 0;
+	BOOL	handled = FALSE;
+	if(translateKey(key, &translated_key))
+	{
+		handled = handleTranslatedKeyUp(translated_key, translated_mask);
+	}
+    if (!handled)
+    {
+        LLGameControl::onKeyUp(translated_key, translated_mask);
+    }
+
+	return handled;
+}
```

Comment:
These have been moved from `LLKeyboard`--derived classes into the base class.   After a little cleanup the logic pattern was identical acrossl platforms.